src/pages/storage/StorageBucketKeys.tsx

```diff
@@ -0,0 +1,295 @@
+  window.innerWidth <= collapsedViewMaxWidth;
+
+interface Props {
+  bucket: LxdStorageBucket;
+}
+
+const StorageBucketKeys: FC<Props> = ({ bucket }) => {
+  const docBaseLink = useDocs();
+  const [query, setQuery] = useState<string>("");
+  const notify = useNotify();
+  const [selectedNames, setSelectedNames] = useState<string[]>([]);
+  const [processingNames, setProcessingNames] = useState<string[]>([]);
+  const [isSmallScreen, setSmallScreen] = useState(figureCollapsedScreen());
+  const { project } = useCurrentProject();
+  const { data: keys = [] } = useBucketKeys(bucket, project?.name ?? "");
+
+  useEffect(() => {
+    const validNames = new Set(keys?.map((key) => key.name));
+    const validSelections = selectedNames.filter((name) =>
+      validNames.has(name),
+    );
+    if (validSelections.length !== selectedNames.length) {
+      setSelectedNames(validSelections);
+    }
+  }, [keys]);
+
+  const filteredKeys =
+    keys?.filter((key) => {
+      if (query) {
+        if (!key.name.toLowerCase().includes(query.toLowerCase())) {
+          return false;
+        }
+        if (!key.description.toLowerCase().includes(query.toLowerCase())) {
+          return false;
+        }
+        if (!key["access-key"].toLowerCase().includes(query.toLowerCase())) {
+          return false;
+        }
+        if (!key["secret-key"].toLowerCase().includes(query.toLowerCase())) {
+          return false;
+        }
+      }
+      return true;
+    }) ?? [];
+
+  const hasKeys = keys && keys.length > 0;
+
+  const headers = [
+    {
+      content: isSmallScreen ? (
+        <>
+          Name
+          <br />
+          <div className="header-second-row">Role</div>
+        </>
+      ) : (
+        "Name"
+      ),
+      sortKey: isSmallScreen ? "role" : "name",
+      className: "name",
+    },
+    {
+      content: "Role",
+      sortKey: "role",
+      className: "role",
+    },
+    ...(isSmallScreen
+      ? []
+      : [
+          {
+            content: "Description",
+            sortKey: "description",
+            className: "description",
+          },
+        ]),
+    { content: "Access key", className: "key-field", sortKey: "access-key" },
+    { content: "Secret key", className: "key-field", sortKey: "secret-key" },
+    { "aria-label": "Actions", className: "actions" },
+  ];
+
+  const rows = filteredKeys.map((key) => {
+    return {
+      key: key.name,
+      className: "u-row",
+      name: key.name,
+      columns: [
+        {
+          content: (
+            <>
+              <div className="u-truncate" title={`Key ${key.name}`}>
+                <ItemName item={key} />
+              </div>
+              {isSmallScreen && key.role}
+            </>
+          ),
+          role: "rowheader",
+          "aria-label": "Name",
+          className: "name",
+        },
+        ...(isSmallScreen
+          ? []
+          : [
+              {
+                content: key.role,
```

Comment:
Here and in the collapsed above, we should call `capitalizeFirstLetter`. So the presentation is the same as in the edit/create form.
```suggestion
                content: capitalizeFirstLetter(key.role),
```